pkg/providers/cloudstack/validator.go

```diff
@@ -302,7 +301,7 @@ func (v *Validator) setDefaultAndValidateControlPlaneHostPort(cloudStackClusterS
 	if !networkutils.IsPortValid(port) {
 		return fmt.Errorf("host %s has an invalid port", pHost)
 	}
-	if !v.skipIpCheck && networkutils.IsPortInUse(v.netClient, host, port) {
+	if !skipIpCheck && networkutils.IsPortInUse(v.netClient, host, port) {
```

Comment:
>And then call it from SetupAndValidateUpgradeCluster but not from SetupAndValidateCreateCluster.

To be clear, you mean the inverse, right? I'd like to check this IP is unused only on creation, **not** on upgrade

Splitting it into a separate public method is a bit annoying because then you have to check the address' formatting twice to be able to separate the host and the port, or else assume that this defaulting and formatting function has already been called. Here it's already split up so you can directly proceed to call the network util function without any extra validation